Automattic/wp-calypso
# State: Introduce site connection actions, reducer, selectors and query component

This PR introduces actions, reducer, selectors and query components for site connection. It consists of:

* `requestConnectionStatus` action - used to fetch the connection status
* `connection` reducer - used to handle connection data and requests
* `getSiteConnectionStatus` selector - used to retrieve the connection status from the state
* `isRequestingSiteConnectionStatus` selector - whether currently retrieving the connection status
* `QuerySiteConnectionStatus` query component - used to handle network requests for fetching the connection status.
* Tests for all of the above.

These will be used to replace the `JetpackSite.prototype.callHome` method and the attributes that it manages (`callingHome` and `unreachable`) as we migrate away from `sites-list`. 

To test:
* Checkout this branch
* Verify all tests pass:
  * `npm run test-client client/state/sites/connection/`
  * `npm run test-client client/state/selectors/test/get-site-connection-status.js`
  * `npm run test-client client/state/selectors/test/is-requesting-site-connection-status.js`
* Insert the `<QuerySiteConnectionStatus siteId={ siteId } />` component somewhere and verify it performs the expected network requests.
